webview-ui/src/components/cloud/UsagePreview.tsx

```diff
@@ -0,0 +1,194 @@
+import { useEffect, useState } from "react"
+import { VSCodeProgressRing } from "@vscode/webview-ui-toolkit/react"
+import { CircleAlert, SquareArrowOutUpRight } from "lucide-react"
+import { useAppTranslation } from "@src/i18n/TranslationContext"
+import { vscode } from "@src/utils/vscode"
+import { formatDateShort } from "@/utils/format"
+
+interface DailyUsage {
+	date: string // ISO date string
+	taskCount: number
+	tokenCount: number
+	cost: number // in USD
+}
+
+interface UsageStats {
+	days: DailyUsage[]
+	totals: {
+		tasks: number
+		tokens: number
+		cost: number
+	}
+}
+
+interface UsagePreviewProps {
+	onViewDetails: () => void
+}
+
+export const UsagePreview = ({ onViewDetails }: UsagePreviewProps) => {
+	const { t } = useAppTranslation()
+	const [isLoading, setIsLoading] = useState(true)
+	const [error, setError] = useState<string | null>(null)
+	const [data, setData] = useState<UsageStats | null>(null)
+
+	// Format large numbers (e.g., 14800000 -> "14.8M")
```

Comment:
Do we already have something like this for tokens? Might be nice to reuse (and also to internationalize)